public/components/common/custom-search-bar/custom-search-bar.tsx

```diff
@@ -1,165 +1,177 @@
+    filterManager.removeAll();
+    filterManager.addFilters(currentFilters);
+    const customFilter = buildCustomFilter(false, values);
+    filterManager.addFilters(customFilter);
+  };
+
+  const refreshCustomSelectedFilter = () => {
+    setSelectedOptions(defaultSelectedOptions);
+    const filters =
+      filterManager
+        .getFilters()
+        .filter(
+          (item) =>
+            item.meta.type === 'phrases' || Object.keys(selectedOptions).includes(item.meta.key)
+        )
+        .map((element) => ({ params: element.meta.params, key: element.meta.key })) || [];
+   
+    const getFilterCustom = (item) => {
+      return item.params.map((element) => ({ label: element, value: item.key }));
+    };
+    const filterCustom = filters.map((item) => getFilterCustom(item)) || [];
+    if (filterCustom.length != 0) {
+      filterCustom.forEach((item) => {
+        item.forEach(element =>{
+            setSelectedOptions((prevState) => ({
+                ...prevState,
+                [element.value]: [...prevState[element.value], element],
+              }));
+        })
+      });
     }
-
-    return (
-        <>
-            <EuiFlexGroup className='custom-kbn-search-bar' alignItems='center' style={{ margin: '0 8px' }}>
-                {
-                    avancedFiltersState === false ?
-                        filtersValues.map((item, key) => (
-                            <EuiFlexItem key={key}>
-                                {getComponent(item)}
-                            </EuiFlexItem>
-                        ))
-                        :
-                        ''
-                }
-                <EuiFlexItem>
-                    <KbnSearchBar
-                        showFilterBar={false}
-                        showQueryInput={avancedFiltersState}
-                        onFiltersUpdated={onFiltersUpdated}
-                    />
-                </EuiFlexItem>
-            </EuiFlexGroup>
-            <EuiFlexGroup justifyContent='flexEnd' style={{ margin: '0 20px' }}>
-                <EuiFlexItem className={'filters-search-bar'} style={{ margin: '0px' }}>
-                    <KbnSearchBar
-                        showDatePicker={false}
-                        showQueryInput={false}
-                        onFiltersUpdated={onFiltersUpdated}
-                    />
-                </EuiFlexItem>
-                <EuiFlexItem grow={false}>
-                    <EuiSwitch
-                        label="Advanced filters"
-                        checked={avancedFiltersState}
-                        onChange={() => changeSwitch()}
-                    />
-                </EuiFlexItem>
-            </EuiFlexGroup>
-        </>
-    )
-};
\ No newline at end of file
+  };
+
+  const onChange = (values: any[]) => {
+    setKibanaFilters(values);
+    refreshCustomSelectedFilter();
+  };
+
+  const getComponent = (item: any) => {
+    var types = {
```

Comment:
**nitpick**: You could use `const` instead of `var` here. Anyways, we prefer `let` over `var` in the variable declaration.